sycl/test-e2e/Matrix/Inputs/joint_matrix_half_impl.hpp

```diff
@@ -5,26 +5,31 @@
 // SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 //
 //===----------------------------------------------------------------------===//
-template <typename T, size_t TM, size_t TN, size_t TK> class mult;
+template <typename T, typename TAcc, typename TResult, size_t TM, size_t TN,
```

Comment:
this test is also almost identical to previous 2 tests. The only difference is bfloat16 -> half, and this type is already a template parameter. I think we should get rid of this duplication and have something like:
`joint_matrix_16bit_impl.hpp`, which would cover both bfloat16 and half kernels, and both packed and row_major B layout. We can have different .cpp files to split tests to multiple runs for different shapes, etc... as we see fit.